### DISCUSSION TITLE
extract table from given elements

### DISCUSSION THREAD

**Discussion by Eyderoe** [jsvine/pdfplumber#1162] [Ideas]
extract table from given elements(special lines, rects, or somthing use to extract table now )
extract table only accept setting now, and other process all do by pdfplumber.
and in example below, red line circle 1 column splite to 2, the extract effect is not good
![a1](https://github.com/jsvine/pdfplumber/assets/103725178/ea6e0e4d-c377-40bd-ba66-85f2c57ecfb2)
although i can ident the right edges (mark with blue), but there's no function that only use the elements (mark in blue) to form a table
is possible to achieve it ? (I thought its easy to achieve and more flexible)
sincerely, Eyderoe
![6785](https://github.com/jsvine/pdfplumber/assets/103725178/581276a9-2b1c-4753-a2b1-07effa47da9d)

**Comment by jsvine** [comment]
Interesting example, @Eyderoe; thanks for sharing. There's no prepackaged table-extraction setting that would address this — to `pdfplumber`, that layout looks like a single table. If there's a programmatic/logic-based way to reliably identify the division of these two tables, then you could use `page.crop(...)` or `page.within_bbox(...)` (depending on the specific structure of the PDF), to focus on the two parts separately and then use `.extract_table(...)` on those two parts individually.

**Comment by Eyderoe** [comment]
thanks, I'll try and feedback laaaater
